packages/lit-renderer/src/lit-renderer.js

```diff
@@ -0,0 +1,98 @@
+import { nothing } from 'lit';
+import { AsyncDirective } from 'lit/async-directive.js';
+import { PartType } from 'lit/directive.js';
+
+/**
+ * @typedef {import('lit/directive.js').PartInfo} PartInfo
+ * @typedef {import('lit/directive.js').ElementPart} ElementPart
+ */
+
+// A symbol that indicates that the renderer hasn't been initialized.
+const VALUE_NOT_INITIALIZED = Symbol('valueNotInitialized');
+
+export class LitRendererDirective extends AsyncDirective {
+  /**
+   * @type {unknown}
+   */
+  previousValue = VALUE_NOT_INITIALIZED;
+
+  /**
+   * @param {PartInfo} part
+   */
+  constructor(part) {
+    super(part);
+    if (part.type !== PartType.ELEMENT) {
+      // TODO: Improve the error message by mentioning the actual name of the renderer directive.
+      throw new Error('Renderer can be only bound to an element.');
+    }
+  }
+
+  /**
+   * @param {ElementPart} part
+   * @param {[R, unknown]}
+   * @return {unknown}
+   */
+  update(part, [renderer, value]) {
+    const firstRender = this.previousValue === VALUE_NOT_INITIALIZED;
+
+    if (!this.__hasChanged(value)) {
+      return nothing;
+    }
+
+    this.host = part.options?.host;
```

Comment:
Let's not use optional chaining as we still need to support webpack 4 in Flow, without using Babel.
While support for `?.` landed in `acorn` [7.3.0](https://github.com/acornjs/acorn/releases/tag/7.3.0), webpack 4 still uses an old [`6.1.4`](https://github.com/webpack/webpack/blob/3956274f1eada621e105208dcab4608883cdfdb2/package.json#L12) version.

Reply:
Done.